### PR TITLE
Move Revenue table to its own component

### PR DESCRIPTION
**Pull request by Aljullu**
Move _Revenue_ report table to its own component, in a similar way to how it's done in _Orders_ and _Products_ reports.

### Screenshots
_(this PR shouldn't produce any visible change)_
image

### Detailed test instructions:

- Go to the _Revenue_ analytics page (`/wp-admin/admin.php?page=wc-admin#/analytics/revenue`).
- Test the table: sorting, using pagination, downloading...
- Verify there are no regressions.

### CODE REVIEW

File: client/analytics/report/revenue/chart.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import PropTypes from 'prop-types';
 import ReportChart from 'analytics/components/report-chart';
 import ReportSummary from 'analytics/components/report-summary';
 
-class OrdersReportChart extends Component {
+class RevenueReportChart extends Component {
```
Comment (timmyc) [inline client/analytics/report/revenue/chart.js]: ahhh nice copy/paste cleanup!

File: client/analytics/report/revenue/table.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+import { Card, Link, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { downloadCSVFile, generateCSVDataFromTable, generateCSVFileName } from 'lib/csv';
+import { formatCurrency, getCurrencyFormatDecimal } from 'lib/currency';
+import { getDateFormatsForInterval, getIntervalForQuery } from 'lib/date';
+import { onQueryChange } from 'lib/nav-utils';
+import ReportError from 'analytics/components/report-error';
+
+export default class RevenueReportTable extends Component {
+	constructor( props ) {
+		super( props );
```
Comment (timmyc) [inline client/analytics/report/revenue/table.js]: Do we need this here?

**Review by timmyc — APPROVED** [review]
Thanks for doing this refactor 🙌

File: client/analytics/report/revenue/table.js (hunk 2 above)
Comment (Aljullu) [reply client/analytics/report/revenue/table.js]: Actually we don't, good catch! :+1:

**Pull request merged**